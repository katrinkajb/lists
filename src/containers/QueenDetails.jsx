import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../components/app/App.css';
import { getQueenById } from '../services/dragRaceAPI';

export default class QueenDetails extends Component {
    static propTypes = {
        match: PropTypes.shape({
            params: PropTypes.shape({
                id: PropTypes.number.isRequired
            }).isRequired
        }).isRequired
    }

    state={
        queen: {
            id: 0,
            name: '',
            winner: false,
            missCongeniality: false,
            quote: '',
            // seasons: [],
            // lipsyncs: [],
            image_url: ''
        },
    }

    async componentDidMount() {
        const oneQueen = await getQueenById(this.props.match.params.id);
        await this.setState({
            queen: oneQueen
        });
    }

    render() {
        return (
            <>
                <h1>{this.state.queen.name}</h1>
                <div>{this.state.queen.winner}</div>
                <div>{this.state.queen.missCongeniality}</div>
                <div>"{this.state.queen.quote}"</div>
                {/* {this.state.queen.seasons}
                {this.state.queen.lipsyncs} */}
                <img 
                    src={this.state.queen.image_url}
                    alt={this.state.queen.name} 
                    style={{
                        height: '500px',
                        width: 'auto',
                    }} 
                />
            </>
        );
    };
}

