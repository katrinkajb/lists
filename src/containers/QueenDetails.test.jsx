import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import QueenDetails from './QueenDetails';

describe('QueenDetails', () => {
    afterEach(() => cleanup());

    it('shows details for a drag queen', async () => {
        render(
            <QueenDetails
                match={{
                    params: {
                    id: 12,
                    },
                }}
            />
        );

        const header = await screen.findByText('Bebe Zahara Benet');

        expect(header).toMatchSnapshot();
    });
})
