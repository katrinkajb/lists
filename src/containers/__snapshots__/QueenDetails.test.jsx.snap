// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueenDetails shows details for a drag queen 1`] = `
<h1>
  Bebe Zahara Benet
</h1>
`;
